src/Common/GuardedPoolAllocatorCrashHandler.h

```diff
@@ -0,0 +1,98 @@
+#pragma once
+
+#include <Common/GuardedPoolAllocatorCommon.h>
+
+#include <Poco/Logger.h>
+
+namespace clickhouse_gwp_asan
+{
+
+class CrashReporter
+{
+public:
+    CrashReporter(
+        const clickhouse_gwp_asan::GuardedPoolAllocatorState * allocator_state_,
```

Comment:
You don't need to use qualified name here, i.e. namespace can be omitted, since you are in the same namespace.